$OpenBSD: patch-dbdimp_c,v 1.1 2012/03/09 13:46:05 pea Exp $

Fix for format string security issue with a malicious server
See https://rt.cpan.org/Public/Bug/Display.html?id=75642

--- dbdimp.c.orig	Fri Mar  9 14:29:19 2012
+++ dbdimp.c	Fri Mar  9 14:30:16 2012
@@ -328,7 +328,7 @@ static void pg_warn (void * arg, const char * message)
 						DBIc_is(imp_dbh, DBIcf_PrintWarn) ? 1 : 0);
 
 		if (DBIc_WARN(imp_dbh) && DBIc_is(imp_dbh, DBIcf_PrintWarn))
-			warn(message);
+			warn("%s", message);
 
 		if (TEND) TRC(DBILOGFP, "%sEnd pg_warn\n", THEADER);
 	}
@@ -1531,7 +1531,7 @@ int dbd_st_prepare (SV * sth, imp_sth_t * imp_sth, cha
 
 		if (pg_st_prepare_statement(aTHX_ sth, imp_sth)!=0) {
 			TRACE_PQERRORMESSAGE;
-			croak (PQerrorMessage(imp_dbh->conn));
+			croak ("%s", PQerrorMessage(imp_dbh->conn));
 		}
 	}
 
